code/modules/cm_marines/overwatch.dm

```diff
@@ -1065,6 +489,384 @@
+	"medics" = squad.num_medics,
+	"smartgun" = squad.num_smartgun,
+	"specialists" = squad.num_specialists,
+	"rto" = squad.num_rto,
+	"total_deployed" = squad.count,
+	"sl_candidates" = squad_leader_candidates,
+	"alive" = alive_squad_members,
+	)
+
+	return(squad_data)
+
+/obj/structure/machinery/computer/overwatch/proc/get_marine_data(datum/squad/squad)
+	var/list/marines_list = squad.marines_list
+
+	for(var/mob/living/carbon/human/current_squaddie in marines_list)
+
+		var/manually_filtered
+		var/marine_status
+		var/filtered
+		switch(current_squaddie.stat)
+			if(CONSCIOUS)
+				marine_status = "Conscious"
+			if(UNCONSCIOUS)
+				marine_status = "Unconscious"
+			if(DEAD)
+				marine_status = "Dead"
+				if(dead_hidden)
+					filtered = TRUE
+
+		var/turf/current_turf = get_turf(current_squaddie)
+		switch(z_hidden)
+			if(HIDE_ALMAYER)
+				if(is_mainship_level(current_turf.z))
+					filtered = TRUE
+			if(HIDE_GROUND)
+				if(is_ground_level(current_turf.z))
+					filtered = TRUE
+
+		if("\ref[current_squaddie]" in marine_filter)
+			manually_filtered = TRUE
+			if(marine_filter_enabled)
+				filtered = TRUE
+
+		var/area/current_area = get_area(current_squaddie)
+		marines_list[current_squaddie] = list(
+			"ref" = ref(current_squaddie),
+			"name" = current_squaddie.real_name,
+			"mob_state" = marine_status,
+			"role" = current_squaddie.job ? current_squaddie.job : current_squaddie?.wear_id.rank,
+			"act_sl" = current_squaddie.job != JOB_SQUAD_LEADER && squad.squad_leader == current_squaddie,
+			"fteam" = current_squaddie?.assigned_fireteam,
+			"dist" = squad.squad_leader ? current_squaddie != squad.squad_leader ? get_dist(current_squaddie, current_squad.squad_leader) : null : null,
+			"dir" = squad.squad_leader ? current_squaddie != squad.squad_leader ? dir2text_short(get_dir(current_squad.squad_leader, current_squaddie)) : null : null,
+			"area_name" = sanitize_area(current_area?.name),
+			"helmet" = istype(current_squaddie.head, /obj/item/clothing/head/helmet/marine),
+			"filtered" = filtered,
+			"manually_filtered" = manually_filtered,
+			"SSD" = !current_squaddie.key || !current_squaddie.client && current_squaddie.stat != DEAD
+			)
+
+	return(marines_list)
+
+/obj/structure/machinery/computer/overwatch/ui_status(mob/user, datum/ui_state/state)
+	. = ..()
+	if(inoperable())
+		return UI_CLOSE
+
+/obj/structure/machinery/computer/overwatch/ui_state(mob/user)
+	return GLOB.not_incapacitated_and_adjacent_state
+
+/obj/structure/machinery/computer/overwatch/ui_data(mob/user)
+	var/list/data = list()
+
+	data["z_hidden"] = z_hidden
+	data["x_supply"] = x_supply
+	data["y_supply"] = y_supply
+	data["x_bomb"] = x_bomb
+	data["y_bomb"] = y_bomb
+	data["dead_hidden"] = dead_hidden
+	data["marine_filter"] = marine_filter
+	data["marine_filter_enabled"] = marine_filter_enabled
+	data["operator"] = operator?.name
+	data["bombardment_cooldown"] = COOLDOWN_TIMELEFT(almayer_orbital_cannon, ob_firing_cooldown)
```

Comment:
feed world.time and ob_firing_cooldown into the UI then do the checks in the UI itself rather than in the DMcode